Jenkinsfile

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env groovy
+
+/**
+* Checks if as message contains a given command and machine.
+*
+* @param message The actual message.
+* @param command The command.
+* @param machine The machine.
+* @return A boolean indicating whether there is a match.
+*/
+boolean machineCheck(String message, String command, String machine) {
+    def matchPattern = "@jenkins-cscs\\s+${command}\\s+${machine}"
+    return message ==~ machinePattern
+}
+
+def dirPrefix = 'reframe-ci'
+def loginBash = '#!/bin/bash -l'
+def bashScript = 'ci-scripts/ci-runner.bash'
+def publicSettings = 'reframe/settings.public.py'
```

Comment:
This should be the `reframe/settings.py`.

Reply:
Ok, I see what you are doing. It's the temporary file for swapping the files. We could have then following structure:

```
config/
    generic.py
    cscs.py

reframe/
    settings.py -> ../config/generic.py
```
So later in your Jenkinsfile you can simply change the link to the desired configuration.

Reply:
I do it like that so that the bash script cannot be changed. But I will change the bash script as well.

Reply:
I don't think you need to change anything in the bash script. All you have to do is before invoking the bash script, set up correctly the symlink. You can do that from the `Jenkinsfile`.